partner_property/models/res_company.py

```diff
@@ -55,3 +55,34 @@ def _inverse_partner_properties_definition_person(self):
                 item.partner_properties_definition_person, item
             )
             ICP.sudo().set_param("partner_property.properties_definition_person", value)
+
+    @api.model
+    def web_search_read(
+        self, domain, specification, offset=0, limit=None, order=None, count_limit=None
+    ):
+        """Override the method to return the "appropriate" company if searched
+        by any of the fields.
+        This method is used to display the search fields (Add custom filter), the
+        properties in the partners are multi-company, and the domain
+        [("partner_properties_definition_company", "!=', False)] would return all
+        the companies, in that case the data of the last company would be displayed,
+        which would be totally confusing.
+        Example:
+        Existing companies: Company A + Company B + Company C
+        Selected company: Company A
+        Now it will show: Property custom (Company A).
+        """
+        f_names = [
+            "partner_properties_definition_company",
+            "partner_properties_definition_person",
+        ]
+        if any(dom[0] in f_names for dom in domain):
+            domain = [("id", "=", self.env.company.id)]
```

Comment:
Have you tried to return an invalid domain for not returning anything at all and have it empty?

```suggestion
            domain = [("id", "=", -1)]
```

Reply:
Yes, I have tried that option and discarded it (it has the side effect of not even showing the properties).

![ejemplo](https://github.com/user-attachments/assets/e36de0d2-409a-4b99-a04f-c4a7609b583e)